src/device-state.ts

```diff
@@ -647,6 +648,7 @@ export async function reboot(force?: boolean, skipLock?: boolean) {
 }
 
 export async function shutdown(force?: boolean, skipLock?: boolean) {
+	await updateLock.ensureNoHUPBreadcrumbsOnHost();
```

Comment:
Adding the HUP checks here instead of in updateLock.lock. That function is currently only entered when there is an application running on the device, because `lockingIfNecessary` is part of `applicationManager`. However when no app / service is present, updateLock.lock never gets called, hence why I was having trouble with integration tests before.